### ISSUE TITLE
Keystore Fails to be Loaded at the Path Listed in orientdb-server-config.xml

### ISSUE THREAD

**Issue by compensator** [orientechnologies/orientdb#4628]
When a Keystore is created and referenced in the orient-server-config.xml file in 2.1-rc5, it fails to be loaded.  I downloaded the source code and debugged this issue.  The problem occurs in OSystemVariableResolver:

```java
public static String resolveVariable(final String variable) {
    if( variable == null )
      return null;
    String resolved = System.getProperty(variable);
    if (resolved == null)
      // TRY TO FIND THE VARIABLE BETWEEN SYSTEM'S ENVIRONMENT PROPERTIES
      resolved = System.getenv(variable);
    return resolved;
}
```
System.getProperty(variable) with variable = "ORIENTDB_HOME" returns "%ORIENTDB_HOME%" on the Windows OS.  In the chain of calls at startup, this returns to the SSLSocketFactory and which attempts to load the keystore from:

"%ORIENTDB_HOME%\config\certs\orientdb.ks"

as set in the orientdb-server-config.xml, which clearly doesn't exist.  The condition should be changed to:
```java
if (resolved != null)
```

**Comment by lvca** [comment]
@compensator Could you elaborate better the problem?

**Comment by compensator** [comment]
I was attempting to load a keystore in 2.0.12 and was receiving the terse error of "invalid or corrupt" keystore despite the fact that the keystore could be loaded via the keytool with the identical values that were in the orientdb-server-config.xml file.  After trying a variety of solution paths I decided to pull down the source code of the latest code (since I need to analyze other items anyway).  I traced through the code and found that an Exception was being thrown in the below OServerSSLSocketFactory.getSSLContext() when it attempts keyStore.load(..).  When I traced through the code the problem was occurring in the following code snippet within OServerSSLSocketFactory where the keyStore path is completed with ORIENTDB_HOME.  What was being returned by OSystemVariableResolver.resolveVariables(${ORIENTDB_HOME}) was "%ORIENTDB_HOME%" rather than, e.g., "D:\orientdb-community-2.0.12\" :

```java
		keyStoreFile = new File(keyStorePath);
		if (!keyStoreFile.isAbsolute()) {
			keyStoreFile = new File(
					OSystemVariableResolver
							.resolveSystemVariables("${ORIENTDB_HOME}"),
					keyStorePath);
		}

		if (trustStorePath != null) {
			trustStoreFile = new File(trustStorePath);
			if (!trustStoreFile.isAbsolute()) {
				trustStoreFile = new File(
						OSystemVariableResolver
								.resolveSystemVariables("${ORIENTDB_HOME}"),
						trustStorePath);
			}
		}
```

```java
	protected SSLContext getSSLContext() {

		try {
			SSLContext context = SSLContext.getInstance("TLS");

			KeyManagerFactory kmf = KeyManagerFactory
					.getInstance(KeyManagerFactory.getDefaultAlgorithm());

			KeyStore keyStore = KeyStore.getInstance(keyStoreType);
			char[] keyStorePass = keyStorePassword.toCharArray();
			keyStore.load(new FileInputStream(keyStoreFile), keyStorePass);

			kmf.init(keyStore, keyStorePass);

			TrustManagerFactory tmf = null;
			if (trustStoreFile != null) {
				tmf = TrustManagerFactory.getInstance(TrustManagerFactory
						.getDefaultAlgorithm());
				KeyStore trustStore = KeyStore.getInstance(trustStoreType);
				char[] trustStorePass = trustStorePassword.toCharArray();
				trustStore.load(new FileInputStream(trustStoreFile),
						trustStorePass);
				tmf.init(trustStore);
			}

			context.init(kmf.getKeyManagers(),
					(tmf == null ? null : tmf.getTrustManagers()), null);

			return context;
		} catch (Exception e) {
			throw new OConfigurationException("Failed to create SSL context", e);
		}
	}

I found that the section of code in the original post was returning "%ORIENTDB_HOME%".  I changed the condition to "!= null" to allow the proper value to return and it worked correctly.

Meanwhile, I copied the orientdb-system-config.xml from the downloaded zip and re-entered the keystore values and it began to load the keystore.  I thus am unsure what the problem was in 2.0.12 vs 2.1-rc5.  I haven't had a change to check the code in 2.0.12 yet.

**Comment by compensator** [comment]
@lvca I'll test this again from scratch and if the error doesn't occur, I'll close it.

**Comment by compensator** [comment]
I finally had a little time yesterday to test out some scenarios on this issue.  It fails to load the key within the keystore if that password differs from the keystore's password.

**Comment by lvca** [comment]
@compensator Please could you send us a PR with the fix?

**Comment by compensator** [comment]
After I reviewed the issue back then in IntelliJ, I blew away the project.  As long as it works, then I don't see the point in pulling down the code and entering that modification.  I am concerned that there will be unintended consequences because other execution paths flow through that section of code.  I also only tested this on Windows.  It may be the case that it works as intended on Linux/Unix.
